File: src/main/java/com/matag/game/status/GameStatusFactory.java
package com.matag.game.status;

import com.matag.game.stack.SpellStack;
import com.matag.game.turn.Turn;
import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.stereotype.Component;

@Component
public class GameStatusFactory implements ApplicationContextAware {
  private ApplicationContext applicationContext;

  @Override
  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
    this.applicationContext = applicationContext;
  }

  public GameStatus create(String gameId) {
    GameStatus gameStatus = new GameStatus(new Turn(), new SpellStack());
    gameStatus.setGameId(gameId);
    return gameStatus;
  }
}

File: src/main/kotlin/com/matag/game/status/GameStatusFactory.kt
package com.matag.game.status

import org.springframework.beans.BeansException
import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationContextAware
import org.springframework.stereotype.Component

@Component
class GameStatusFactory : ApplicationContextAware {
    private var applicationContext: ApplicationContext? = null

    @Throws(BeansException::class)
    override fun setApplicationContext(applicationContext: ApplicationContext) {
        this.applicationContext = applicationContext
    }

    fun create(gameId: String?): GameStatus {
        val gameStatus = applicationContext!!.getBean(GameStatus::class.java)
        gameStatus.gameId = gameId
        return gameStatus
    }
}
